metricshub-snmpv3-extension/src/main/java/org/sentrysoftware/metricshub/extension/snmpv3/SnmpV3RequestExecutor.java

```diff
@@ -21,265 +21,33 @@
-
-		String result = executeSNMPGetRequest(
-			SnmpGetRequest.GET,
-			oid,
-			(SnmpV3Configuration) configuration,
+	protected SnmpClient createSnmpClient(ISnmpConfiguration protocol, String hostname) throws IOException {
+		SnmpV3Configuration snmpConfig = (SnmpV3Configuration) protocol;
+		return new SnmpClient(
 			hostname,
+			snmpConfig.getPort(),
+			snmpConfig.getIntVersion(),
+			snmpConfig.getRetryIntervals(),
 			null,
-			logMode
-		);
-
-		final long responseTime = System.currentTimeMillis() - startTime;
-
-		LoggingHelper.trace(() ->
-			log.trace(
-				"Executed SNMP V3 Get request:\n- OID: {}\n- Result: {}\n- response-time: {}\n",
-				oid,
-				result,
-				responseTime
-			)
-		);
-
-		return result;
-	}
-
-	/**
-	 * Execute SNMP Table
-	 *
-	 * @param oid               The SNMP Object Identifier (OID) representing the table.
-	 * @param selectColumnArray An array of column names to select from the SNMP table.
-	 * @param configuration     The SNMP configuration containing connection details.
-	 * @param hostname          The hostname or IP address of the SNMP-enabled device.
-	 * @param logMode           Flag indicating whether to log warnings in case of errors.
-	 * @return A list of rows, where each row is a list of string cells representing the SNMP table.
-	 * @throws InterruptedException If the thread executing this method is interrupted.
-	 * @throws ExecutionException  If an exception occurs during the execution of the SNMP request.
-	 * @throws TimeoutException    If the SNMP request times out.
-	 */
-	@WithSpan("SNMP V3 Get Table")
-	@Override
-	public List<List<String>> executeSNMPTable(
-		@NonNull @SpanAttribute("snmp.oid") final String oid,
-		@NonNull @SpanAttribute("snmp.columns") String[] selectColumnArray,
-		@NonNull @SpanAttribute("snmp.config") final ISnmpConfiguration configuration,
-		@NonNull @SpanAttribute("host.hostname") final String hostname,
-		final boolean logMode
-	) throws InterruptedException, ExecutionException, TimeoutException {
-		LoggingHelper.trace(() ->
-			log.trace("Executing SNMP Table request:\n- OID: {}\n- Columns: {}\n", oid, Arrays.toString(selectColumnArray))
-		);
-
-		final long startTime = System.currentTimeMillis();
-
-		List<List<String>> result = executeSNMPGetRequest(
-			SnmpGetRequest.TABLE,
-			oid,
-			(SnmpV3Configuration) configuration,
-			hostname,
-			selectColumnArray,
-			logMode
-		);
-
-		final long responseTime = System.currentTimeMillis() - startTime;
-
-		LoggingHelper.trace(() ->
-			log.trace(
-				"Executed SNMP Table request:\n- OID: {}\n- Columns: {}\n- Result:\n{}\n- response-time: {}\n",
-				oid,
-				Arrays.toString(selectColumnArray),
-				TextTableHelper.generateTextTable(selectColumnArray, result),
-				responseTime
-			)
+			snmpConfig.getAuthType().toString(),
+			snmpConfig.getUsername(),
+			new String(snmpConfig.getPassword()),
+			snmpConfig.getPrivacy().toString(),
+			new String(snmpConfig.getPrivacyPassword()),
```

Comment:
If `snmpConfig.getPrivacyPassword()` is null we will encounter a NullPointerException.